### ISSUE TITLE
Ansible Lint reports git missing version

### ISSUE THREAD

**Issue by monorailpilot** [ansible/ansible-lint#906]
# Issue Type
- Bug report

# Ansible and Ansible Lint details
ansible 2.8.2
ansible-lint 4.2.0

- ansible installation method: pip
- ansible-lint installation method: pip

# Desired Behaviour

The git module can be used to simply pull the hash of the code checked out, for example:
```yaml
  - name: "Get post deploy code release hash"
    git:
      repo: "."
      dest: "."
      clone: no
      update: no
    connection: "local"
    become: false
    register: git_commit
```
This should not report an error in ansible-lint

# Actual Behaviour (Bug report only)

ansible lint is reporting an error
[401] Git checkouts must contain explicit version
*FILE LOCATION HIDDEN*
Task/Handler: Get post deploy code release hash

**Comment by webknjaz** [comment]
Check against master. We do not support old ones.

**Comment by monorailpilot** [comment]
Same issue with ansible-lint 4.3.0a5.dev11+g5fc9f26

**Comment by webknjaz** [comment]
Not sure if it's reasonable to skip abuses of the git module. FWIW you can just apply `noqa` for your exceptional case.

**Comment by monorailpilot** [comment]
I don't consider it an abuse of the module given there are multiple examples of git without a version parameter in the ansible documentation of using the git module. 

Thank you for the heads up on noqa.

**Comment by webknjaz** [comment]
The point of the linter is to (1) help keep the style consistent and (2) help prevent *common* mistakes. In this case, the rule is enforcing the best practice of pinning the versions of software you work with (just like it's not recommended to use latest with apt/yum/dnf/pip/etc.).

This principle is there to help you have reproducible environments. So that you don't download different unpredictable versions of code when pointing to branches at different points in time. Re-running your playbooks should be predictable, and running them again in a year shouldn't suddenly start relying on different versions of software that may behave differently or have different sets of bugs.

That is why this rule is useful in general and any other uses violate it. But since there are unconventional use-cases, you can use noqa or tag-based rule ignores to facilitate your use-case that doesn't work well with the rule.

When pointing to `.` is hard to statically analyze all the implications and so I don't think that this rule can reliably identify such exceptional cases (from the technical standpoint).

If you have any implementation in mind, you may try sending a PR, of course, but I cannot guarantee that it'd be accepted.
